noxfiles/ci_nox.py

```diff
@@ -0,0 +1,202 @@
+def black(session: nox.Session) -> None:
+    """Run the 'black' style linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "black")
+    else:
+        run_command = ("black", "--check", "src", "tests", "noxfiles")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def isort(session: nox.Session) -> None:
+    """Run the 'isort' import linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "isort")
+    else:
+        run_command = ("isort", "--check-only", "src", "tests", "noxfiles")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def mypy(session: nox.Session) -> None:
+    """Run the 'mypy' static type checker."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "mypy")
+    else:
+        run_command = ("mypy",)
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def pylint(session: nox.Session) -> None:
+    """Run the 'pylint' code linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "pylint")
+    else:
+        run_command = ("pylint", "src", "noxfiles")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def xenon(session: nox.Session) -> None:
+    """Run the 'pylint' code linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "xenon")
+    else:
+        run_command = (
+            "xenon",
+            "src",
+            "--max-absolute B",
+            "--max-modules B",
+            "--max-average A",
+            "--ignore 'data, tests, docs'",
+            "--exclude src/fidesctl/_version.py",
+        )
+    session.run(*run_command, external=True)
+
+
+# Fidesctl Checks
+@nox.session()
+def check_install(session: nox.Session) -> None:
+    """Check that fidesctl is installed."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "check_install")
+    else:
+        run_command = ("fidesctl", *(WITH_TEST_CONFIG), "--version")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def fidesctl(session: nox.Session) -> None:
+    """Run a fidesctl evaluation."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "fidesctl")
+    else:
+        run_command = ("fidesctl", "--local", *(WITH_TEST_CONFIG), "evaluate")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def fidesctl_db_scan(session: nox.Session) -> None:
+    """Scan the fidesctl application database to check for dataset discrepancies."""
+    session.notify("teardown")
+    session.run(*START_APP, external=True)
+    run_command = (
+        *RUN,
+        "fidesctl",
+        *(WITH_TEST_CONFIG),
+        "scan",
+        "dataset",
+        "db",
+        "postgresql+psycopg2://postgres:fidesctl@fidesctl-db:5432/fidesctl_test",
+    )
+    session.run(*run_command, external=True)
+
+
+# Pytest
```

Comment:
It would be nice to have the option to run all the tests at once. Maybe still exclude the external tests though since they can't be run by everyone.

```diff
+ @nox.session()
+ def pytest(session: nox.Session) -> None:
+     """Run all tests except external."""
+     session.notify("teardown")
+     session.run(*START_APP, external=True)
+     run_command = (
+         *RUN_NO_DEPS,
+         "pytest",
+         "-x",
+         "-m",
+         "not external",
+     )
+     session.run(*run_command, external=True)
```

Reply:
I'm going to update the nox session to accept test markers as positional arguments. so you can do `nox -s pytest -- 'not external'` or `nox -s pytest -- unit`